### DISCUSSION TITLE
yq e '.container | del(.field)' - doesn't work as I expect

### DISCUSSION THREAD

**Discussion by pmorch** [mikefarah/yq#790] [Q&A] [answered]
I want to get a field, and remove a subfield from the result. I can't seem to do that in one `eval` but it works fine in two.

Actual behavior differs between 4.4.1 and 4.7.0 and either they're both wrong or I'm wrong :-)

#### test.yaml 
```
container:
  fieldA: keep
  fieldB: keep
  field: delete
```

This I what I'm trying to achieve, and both 4.4.1 and 4.7.0 agree that this works:

```
$ ./yq-4.7.0 e '.container' test.yaml | ./yq-4.7.0 e 'del(.field)' -
fieldA: keep
fieldB: keep
```

But if I try to do it in one `eval`/step, it gives output that I don't expect - and different output for 4.4.1 and 4.7.0:
```
# Why isn't field deleted?
$ ./yq-4.7.0 e '.container | del(.field)' test.yaml 
fieldA: keep
fieldB: keep
field: delete

# Why was this an error?
$ ./yq-4.4.1 e '.container | del(.field)' test.yaml 
Error: Cannot delete nodes from parent of tag !!null
```

What am I doing wrong?

**Comment by mikefarah** [comment, marked as answer]
You're not doing anything wrong - you've found a bug in delete! I think I see what the cause is - and wonder how it hasn't been noticed yet!

Regardless, I'll have a fix out as soon as I can, until then you can do the following:


```
yq e 'del(.container.field) | .container' test.yaml
```

**Reply by mikefarah** [reply]
Raised a bug: https://github.com/mikefarah/yq/issues/793

**Reply by mikefarah** [reply]
Fixed in 4.7.1
